data/serializer/src/main/java/tech/pegasys/teku/api/schema/altair/BeaconStateAltair.java

```diff
@@ -109,8 +113,18 @@ public BeaconStateAltair(
     super(beaconState);
     final tech.pegasys.teku.spec.datastructures.state.beaconstate.versions.altair.BeaconStateAltair
         altair = beaconState.toVersionAltair().orElseThrow();
-    this.previous_epoch_participation = altair.getPreviousEpochParticipation().sszSerialize();
-    this.current_epoch_participation = altair.getCurrentEpochParticipation().sszSerialize();
+    Byte[] previousEpochParticipationArray =
+        altair.getPreviousEpochParticipation().stream()
+            .map(AbstractSszPrimitive::get)
+            .collect(Collectors.toList())
+            .toArray(new Byte[altair.getPreviousEpochParticipation().size()]);
+    this.previous_epoch_participation = ArrayUtils.toPrimitive(previousEpochParticipationArray);
+    Byte[] currentEpochParticipationArray =
```

Comment:
I think I'd simplify the conversion to `byte[]` and reduce the garbage created by doing it the old fashioned way:

```
this.previous_epoch_participation = toByteArray(altair.getPreviousEpochParticipation());
```
and:
```
  private byte[] toByteArray(final SszList<SszByte> byteList) {
    final byte[] array = new byte[byteList.size()];
    for (int i = 0; i < array.length; i++) {
      array[i] = byteList.get(i).get();
    }
    return array;
  }
```

Reply:
interestingly there was a copy/paste error there, so definitely worth the cleanup...